internal/backend/local/backend.go

```diff
@@ -89,20 +90,28 @@ type Local struct {
 
 	// opLock locks operations
 	opLock sync.Mutex
+
+	encryption encryption.StateEncryption
 }
 
 var _ backend.Backend = (*Local)(nil)
 
 // New returns a new initialized local backend.
-func New() *Local {
-	return NewWithBackend(nil)
+func New(enc encryption.StateEncryption) *Local {
+	return &Local{
+		encryption: enc,
+	}
 }
 
 // NewWithBackend returns a new local backend initialized with a
 // dedicated backend for non-enhanced behavior.
-func NewWithBackend(backend backend.Backend) *Local {
+func NewWithBackend(backend backend.Backend, enc encryption.StateEncryption) *Local {
+	if backend == nil && enc == nil {
+		panic("either backend or encryption required for backend.Local initialization")
```

Comment:
tiny tiny nitpick: I think this error would be more comprehensible if it was talking about what has happened, rather than what is required.

I would recommend we do something like:
```suggestion
		panic("backend.Local NewWithBackend called with nil backend and encryption, atleast one is required.")
```